### PR TITLE
CM-342: filter only not assigned bills

### PR DESCRIPTION
**Pull request by jdolkowski**
https://openimis.atlassian.net/browse/CM-342

This PR filters only not assigned bills to payroll.

### CODE REVIEW

**Review by dborowiecki — COMMENTED** [review]
Not sure if that is a valid check. Can you confirm that it cover case when bill was detached from the payroll or when the bill is no longer valid?

**Comment by jdolkowski** [comment]
I provided more strict filtering options.

File: payroll/services.py [hunk 1]
```diff
@@ -128,7 +128,11 @@ def _get_bills_queryset(self, obj_data, included_unpaid):
             date_bill__range=(date_from, date_to),
             subject_type=ContentType.objects.get_for_model(Beneficiary),
             subject_id__in=beneficiary_ids,
-            status__in=[Bill.Status.VALIDATED]
+            status__in=[Bill.Status.VALIDATED],
+        )
+
+        bills_queryset = bills_queryset.filter(
+            Q(payrollbill__isnull=True) | Q(payrollbill__is_deleted=True)
```
Comment (sniedzielski) [inline payroll/services.py:135]: Why payroll__is_deleted=True? 
I think that when we delete payroll, all bills are detached and moreover PayrollBill entity is completely removed. Can you confirm @jdolkowski

**Review by sniedzielski — CHANGES_REQUESTED** [review]
@jdolkowski Please take a look into my comment

File: payroll/services.py (hunk 1 above)
Comment (jdolkowski) [reply payroll/services.py:135]: PayrollBill.objects.filter(payroll=obj_).delete()  --> part from delete payroll

they are HistoryModel so they are set to is_deleted true

Comment (sniedzielski) [reply payroll/services.py:135]: ok, then it is ok

**Review by sniedzielski — APPROVED** [review]

**Pull request merged**